include/alpaka/kernel/TaskKernelCpuOmp2Blocks.hpp

```diff
@@ -38,12 +39,702 @@
+                TSchedule const&)
+            {
+#    if _OPENMP < 200805 // For OpenMP < 3.0 you have to declare the loop index (a signed integer) outside of the loop
+                         // header.
+                std::intmax_t iNumBlocksInGrid(static_cast<std::intmax_t>(numIterations));
+                std::intmax_t i;
+#        pragma omp for nowait
+                for(i = 0; i < iNumBlocksInGrid; ++i)
+#    else
+#        pragma omp for nowait
+                for(TIdx i = 0; i < numIterations; ++i)
+#    endif
+                {
+                    loopBody(i);
+                }
+            }
+        };
+
+        //! Helper trait to check if TSchedule is not a type originating from OmpSchedule trait definition
+        //!
+        //! Is well-formed for those types (not defining the trait), ill-formed otherwise.
+        //! When well-formed, result in a non-void type to not collide with void-based SFINAE also used.
+        //!
+        //! \tparam TSchedule The schedule type.
+        template<typename TSchedule>
+        using HasOmpScheduleNotDefined = std::enable_if_t<!std::is_same<TSchedule, omp::Schedule>::value, char>;
+
+        //! Executor of parallel OpenMP loop with the static schedule
+        //!
+        //! The default implementation is for the kernels that do not set chunk size in any way.
+        //!
+        //! Note: HasOmpScheduleNotDefined<TSchedule> should not be necessary, but is required by some old compilers.
+        //!
+        //! \tparam TKernel The kernel type.
+        //! \tparam TSchedule The schedule type (not necessarily omp::Schedule).
+        template<typename TKernel, typename TSchedule>
+        struct ParallelForImpl<TKernel, TSchedule, omp::Schedule::Static, HasOmpScheduleNotDefined<TSchedule>>
+        {
+            //! Run parallel OpenMP loop
+            //!
+            //! \tparam TLoopBody The loop body functor type.
+            //! \tparam TIdx The index type.
+            //!
+            //! \param loopBody The loop body functor instance, takes iteration index as input.
+            //! \param numIterations The number of loop iterations.
+            template<typename TLoopBody, typename TIdx>
+            ALPAKA_FN_HOST void operator()(
+                TKernel const&,
+                TLoopBody&& loopBody,
+                TIdx const numIterations,
+                TSchedule const&)
+            {
+#    if _OPENMP < 200805 // For OpenMP < 3.0 you have to declare the loop index (a signed integer) outside of the loop
+                         // header.
+                std::intmax_t iNumBlocksInGrid(static_cast<std::intmax_t>(numIterations));
+                std::intmax_t i;
+#        pragma omp for nowait schedule(static)
+                for(i = 0; i < iNumBlocksInGrid; ++i)
+#    else
+#        pragma omp for nowait schedule(static)
+                for(TIdx i = 0; i < numIterations; ++i)
+#    endif
+                {
+                    loopBody(i);
```

Comment:
for the INTEL compiler bug, it could help to create before `#pragma omp for nowait schedule(static)` a copy of the lambda function and then declare the copy explicit private.

Another possible workaround would be to apply the arguments for the functor here instead of somewhere before. Maybe the compiler has problems to move some arguments into the parallel for.

Reply:
I was able to work around the example community.intel.com/t5/Intel-C-Compiler/compiler-2020-1-217-internal-error-assertion-failed-find-assoc/m-p/1227009#M37871 by the following code on hemera.

```C++
template < typename F>
void foo(const F& f)
{
  #pragma omp parallel for schedule(dynamic) collapse(2) 
  for (unsigned int v = 0; v < 0; ++v){
   for (unsigned int t = 0; t < 0; ++t){
      auto x = [&f]( const auto & i){ f(i);};
      x(0);
   }
  }
}

void f() {
 foo( [](const auto &indices_2) {} );
}
```